common/flockfile/src/lib.rs

```diff
@@ -1,135 +1,11 @@
-use log::{debug, warn};
-use nix::fcntl::{flock, FlockArg};
-use std::{
-    fs::{self, File, OpenOptions},
-    io,
-    os::unix::io::AsRawFd,
-    path::{Path, PathBuf},
-};
+#[cfg(unix)]
+mod unix;
 
-#[derive(thiserror::Error, Debug)]
-pub enum FlockfileError {
-    #[error(transparent)]
-    IoError(#[from] std::io::Error),
+#[cfg(unix)]
+pub use unix::*;
 
-    #[error("Couldn't acquire file lock.")]
-    NixError(#[from] nix::Error),
-}
+#[cfg(not(unix))]
```

Comment:
As `unix` cfg feature is a shortcut for `target_family = unix` is this intendent to use `windows` module and possibly break macOs or other platofrms?

Reply:
I used the cfg feature already used for the `users` module. Assuming that an issue here would have already been raised by our Mac devs.

Reply:
Successfully tested `cargo build` and `cargo test` on MacOS with these changes.